test/internal/helpers/Player.java

```diff
@@ -38,11 +38,10 @@ public static Cleanups addItem(String item) {
 
   public static Cleanups addItem(String item, int count) {
     var cleanups = new Cleanups();
-    AdventureResult parsed = AdventureResult.tallyItem(item);
-    for (int i = 0; i < count; i++) {
-      AdventureResult.addResultToList(KoLConstants.inventory, parsed);
-      cleanups.add(() -> AdventureResult.removeResultFromList(KoLConstants.inventory, parsed));
-    }
+    AdventureResult parsed = AdventureResult.tallyItem(item, count, true);
```

Comment:
Not introduced in this PR but one issue I noticed with this cleanup is that it is misleading for tests that remove the item added during the course of execution. A better cleanup would be to count the items before and restore to that count in the cleanup

Reply:
Are there such tests? If so, they can't use this. If not, as soon as the first such test is written, it will be time to fix this. :)

Reply:
All the cleanups I wrote assume you're reverting back to a reset character -- for example, `canEquip` and `setStats` just set the stats back to zero, instead of whatever they were before.

Reply:
If that is the assumption we want, then we actually do want AdventureResult.removeFromList().

Reply:
I've written plenty - testing quests that remove a quest item or whatever. It hasn't had an issue because as ryo says, we just reset the character afterwards. I'm just imagining more elaborate tests in the future - but resetting to zero is probably better. Remove from list sounds good. But we can do this later